editoast/src/views/openapi.rs

```diff
@@ -0,0 +1,671 @@
+    ///         .take_new("paths/A", "paths")
+    ///         .take_new("paths/B/C", "paths")
+    ///         .finish();
+    ///     json!({"paths": {"/A": 12, "/B": {"C/": null}, "B/C/": null}})
+    /// );
+    /// ```
+    #[must_use]
+    #[allow(unused)]
+    pub fn take_new(mut self, new_src_path: &str, old_parent_path: &str) -> Self {
+        let src = Path::new(new_src_path);
+        let (new_object, new_key) =
+            find_json_at(src, &mut self.new).expect("take_new: src object must exist");
+        let dst = Path::new(old_parent_path);
+        let dst_object = find_json_at(dst, &mut self.old)
+            .expect("take_new: old_parent_path must exist")
+            .0;
+        let dst_map = dst_object
+            .as_object_mut()
+            .expect("take_new: old_parent_path must be an object");
+        if let Some((_, old_key)) =
+            find_json_at(Path::new(&new_key), &mut Json::Object(dst_map.to_owned()))
+        {
+            dst_map.remove(&old_key);
+        }
+        dst_map.insert(new_key, new_object.clone());
+        self
+    }
+
+    /// Takes the subtree at `new_src_path` in the new openapi and puts it at
+    /// the key `old_parent_path` in the old openapi
+    ///
+    /// ```
+    /// let old = json!({"paths": {"/A": null, "/B": {"C/": null}}});
+    /// let new = json!({"paths": {"/A": 12, "/B": null, "/B/C": 42}});
+    /// assert_eq!(
+    ///     OpenApiMerger::new(old.to_string(), new.to_string())
+    ///         .put_at("paths/A", "paths/A")
+    ///         .put_at("paths/B/C", "paths/B/C")
+    ///         .finish();
+    ///     json!({"paths": {"/A": 12, "/B": {"C/": 42}}})
+    /// );
+    /// ```
+    ///
+    /// See [OpenApiMerger::replace]
+    #[must_use]
+    #[allow(unused)]
+    pub fn put_at(mut self, new_src_path: &str, old_dst_path: &str) -> Self {
+        let src = Path::new(new_src_path);
+        let (new_object, _) =
+            find_json_at(src, &mut self.new).expect("put_at: src object must exist");
+        let dst = Path::new(old_dst_path);
+        let (parent_object, old_key) =
+            find_parent_of(dst, &mut self.old).expect("put_at: old_dst_path parent must exist");
+        parent_object
+            .as_object_mut()
+            .expect("put_at: old_dst_path parent must be an object")
+            .insert(
+                old_key.unwrap_or_else(|| {
+                    dst.file_name()
+                        .expect("put_at: dst path should end with a key name")
+                        .to_str()
+                        .unwrap()
+                        .to_owned()
+                }),
+                new_object.clone(),
+            );
+        self
+    }
+
+    /// Just like [Self::put_at] for when the src and the dst are identical
+    #[must_use]
+    #[allow(unused)]
+    pub fn replace(self, path: &str) -> Self {
+        self.put_at(path, path)
+    }
+
+    #[must_use]
+    #[allow(unused)]
```

Comment:
is it really still unused?

Reply:
Yes, it will be used once we start annotating the endpoints.